plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/Base85Codec.java

```diff
@@ -72,9 +73,9 @@ public static ByteBuffer decode(String encoded)
             word += decodeInputChar(encoded.charAt(inputIndex + 2)) * BASE_2ND_POWER;
             word += decodeInputChar(encoded.charAt(inputIndex + 3)) * BASE;
             word += decodeInputChar(encoded.charAt(inputIndex + 4));
-            buffer.putInt(word);
+            int destination = (inputIndex / 5) * 4;
+            arraycopy(Ints.toByteArray(word), 0, buffer, destination, 4);
```

Comment:
constructing 4-element byte array looks like overhead

so either inline this:

```java
byte[] buffer = new byte[encoded.length() / 5 * 4];

int outputIndex = 0;
for (...) {
...

    buffer[outputIndex] = (byte) (word >> 24);
    //noinspection NumericCastThatLosesPrecision
    buffer[outputIndex + 1] = (byte) (word >> 16);
    //noinspection NumericCastThatLosesPrecision
    buffer[outputIndex + 2] = (byte) (word >> 8);
    //noinspection NumericCastThatLosesPrecision
    buffer[outputIndex + 3] = (byte) word;
    outputIndex += 4;
}
checkState(outputIndex == buffer.length);
return buffer;
```

or use `ByteBuffer` internally to put the Int:

```java
byte[] result = new byte[encoded.length() / 5 * 4];
ByteBuffer buffer = ByteBuffer.wrap(result).order(ByteOrder.LITTLE_ENDIAN);

for (...) {
   ...
    buffer.putInt(word);
}
checkState(buffer.position() == result.length);
return result;
```